lib/src/main/java/graphql/nadel/engine/transform/query/NadelQueryTransformer.kt

```diff
@@ -168,8 +168,14 @@ class NadelQueryTransformer private constructor(
     }
 
     private fun getUnderlyingTypeNames(objectTypeNames: Collection<String>): List<String> {
-        return objectTypeNames.map {
-            executionBlueprint.getUnderlyingTypeName(service, overallTypeName = it)
+        return if (executionContext.hints.sharedTypeRenames(service)) {
+            objectTypeNames.map {
+                executionBlueprint.getUnderlyingTypeName(overallTypeName = it)
+            }
+        } else {
+            objectTypeNames.map {
+                executionBlueprint.getUnderlyingTypeName(service, overallTypeName = it)
```

Comment:
ok, so if something is renamed but the type is declared and renamed in another service
`executionBlueprint.getUnderlyingTypeName(service, overallTypeName = it)` will not actually get you the underlying name as it's only looking at the types owned by `service`. this thing `executionBlueprint.getUnderlyingTypeName(overallTypeName = it)` looks at the whole overall schema tho, for every rename, there can only be 1 underlying name and 1 overall name, so it's always 1 to 1 mapping